src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -116,7 +117,7 @@ export class DeviceDetails extends Component {
               <Accordion title={t('devices.details.tags.title')} description={t('devices.details.tags.description')}>
                 <Grid>
                   {
-                    (device.tags && device.tags.length) &&
+                    (device.tags && Object.keys(device.tags).length > 0) &&
```

Comment:
I would simplify this and the below code a bit. Before the return, call `const tags = Object.keys(device.tags || {})` and save the result. Then you can use the result here and to loop through the results later on.